packages/dev/core/src/Animations/animation.optimizations.ts

```diff
@@ -0,0 +1,67 @@
+import type { Scene } from "core/scene";
+
+/**
+ * Interface used to define the optimization options for animations
+ */
+export interface IAnimationOptimization {
+    /**
+     * If true, all runtime animations will be merged into the first animatable
+     * @defaultValue true
+     */
+    mergeRuntimeAnimations: boolean;
+    /**
+     * If true, all keyframes evaluation will be merged from the first runtime animation
+     * You need to turn on `mergeRuntimeAnimations` for this to work
+     * @defaultValue false
+     */
+    mergeKeyFrames: boolean;
+}

```

Comment:
If the intent is that `mergeKeyFrames` should only be defined if `mergeRuntimeAnimations` is `true`, you can express this with a discriminated union and it will be a compile error when someone doesn't read the doc comments and tries to just set `mergeKeyFrames` to true (which you know will happen :)).
```suggestion
export type AnimationOptimization = {
    /**
     * Do not merge runtime animations
     * @defaultValue true
     */
    mergeRuntimeAnimations: false;
} | {
    /**
     * All runtime animations will be merged into the first animatable
     * @defaultValue true
     */
    mergeRuntimeAnimations: true;
     /**
     * If true, all keyframes evaluation will be merged from the first runtime animation
     * You need to turn on `mergeRuntimeAnimations` for this to work
     * @defaultValue false
     */
    mergeKeyFrames: boolean;
}
```